core/web/solana_transfer_controller.go

```diff
@@ -0,0 +1,140 @@
+	solanaChains := tc.App.GetChains().Solana
+	if solanaChains == nil {
+		jsonAPIError(c, http.StatusBadRequest, ErrSolanaNotEnabled)
+		return
+	}
+
+	var tr solanamodels.SendRequest
+	if err := c.ShouldBindJSON(&tr); err != nil {
+		jsonAPIError(c, http.StatusBadRequest, err)
+		return
+	}
+	if tr.SolanaChainID == "" {
+		jsonAPIError(c, http.StatusBadRequest, errors.New("missing solanaChainID"))
+		return
+	}
+	chain, err := solanaChains.Chain(c.Request.Context(), tr.SolanaChainID)
+	switch err {
+	case solana.ErrChainIDInvalid, solana.ErrChainIDEmpty:
+		jsonAPIError(c, http.StatusBadRequest, err)
+		return
+	case nil:
+		break
+	default:
+		jsonAPIError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	if tr.From.IsZero() {
+		jsonAPIError(c, http.StatusUnprocessableEntity, errors.Errorf("source address is missing: %v", tr.From))
+		return
+	}
+
+	if tr.Amount == 0 {
+		jsonAPIError(c, http.StatusBadRequest, errors.New("amount must be greater than zero"))
+		return
+	}
+
+	fromKey, err := tc.App.GetKeyStore().Solana().Get(tr.From.String())
+	if err != nil {
+		jsonAPIError(c, http.StatusUnprocessableEntity, errors.Errorf("fail to get key: %v", err))
+		return
+	}
+
+	txm := chain.TxManager()
+	var reader client.Reader
+	reader, err = chain.Reader("")
+	if err != nil {
+		jsonAPIError(c, http.StatusInternalServerError, errors.Errorf("chain unreachable: %v", err))
+		return
+	}
+
+	if !tr.AllowHigherAmounts {
+		err = solanaValidateBalance(reader, tr.From, tr.Amount)
+		if err != nil {
+			jsonAPIError(c, http.StatusUnprocessableEntity, errors.Errorf("failed to validate balance: %v", err))
+			return
+		}
+	}
+
+	blockhash, err := reader.LatestBlockhash()
+	if err != nil {
+		jsonAPIError(c, http.StatusInternalServerError, errors.Errorf("failed to get latest block hash: %v", err))
+		return
+	}
+
+	tx, err := solanaGo.NewTransaction(
+		[]solanaGo.Instruction{
+			system.NewTransferInstruction(
+				tr.Amount,
+				tr.From,
+				tr.To,
+			).Build(),
+		},
+		blockhash.Value.Blockhash,
+		solanaGo.TransactionPayer(tr.From),
+	)
+	if err != nil {
+		jsonAPIError(c, http.StatusInternalServerError, errors.Errorf("failed to create tx: %v", err))
+		return
+	}
+
+	_, err = tx.Sign(
+		func(key solanaGo.PublicKey) *solanaGo.PrivateKey {
+			if tr.From.Equals(key) {
+				r := solanaGo.PrivateKey(fromKey.Raw())
+				return &r
+			}
+			return nil
+		},
+	)
+	if err != nil {
+		jsonAPIError(c, http.StatusInternalServerError, errors.Errorf("failed to sign tx: %v", err))
+		return
+	}
+
+	err = txm.Enqueue("", tx)
+	if err != nil {
+		jsonAPIError(c, http.StatusInternalServerError, errors.Errorf("transaction failed: %v", err))
+		return
+	}
+
+	jsonAPIResponse(c, map[string]interface{}{"message": "transaction sent"}, "solana_tx")
+}
+
+func solanaValidateBalance(reader client.Reader, from solanaGo.PublicKey, amount uint64) error {
+	balance, err := reader.Balance(from)
+	if err != nil {
+		return err
+	}
+
+	if balance < amount {
```

Comment:
@aalu1418 @archseer How do fees work? Is there some example code?

Reply:
tx fees are usually calculated based on the number of signers on the transaction. it is currently 5000 lamports per signature on the tx (so in this case it would be 1 sig). that value is demand driven...but it also hasn't changed yet (and we hard code it in the program)

would expose a get fee method in the client that wraps this if that helps? https://docs.solana.com/developing/clients/jsonrpc-api#getfeeformessage

Reply:
added it here: https://github.com/smartcontractkit/chainlink/pull/6231/commits/8c991115f4c94becc25d03c58c4c436c71a7b068